### PR TITLE
Add script to fix markdown table formatting

### PR DESCRIPTION
**Pull request by wanLo**
A Python script that takes XWiki-Markdown 1.2 code and corrects formatting issues within tables by clearing out line breaks that harm the table layout

### CODE REVIEW

**Comment by marcelklehr** [comment]
Looks good.

What happens if I input 

```
|-|-|-|
|foo|

bar after some line breaks|alright|
|bla|bla|bla|
```

Does this occur at all (cells starting with line breaks)? Probably not that important though :man_shrugging:

File: md_format_tables.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/python
+
+# files to be replaced with REST-Requests
+file1 = open('table.txt')
```
Comment (zonradkuse) [inline md_format_tables.py]: It could make sense to separate this completely from the actual logic and having the logic imported later. It would make it easier to write unit tests - for example for Marcel's edge case

Comment (zonradkuse) [reply md_format_tables.py]: line 31 would return the md_output instead of writing it to a file

File: md_format_tables.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+lines = md_input.split('\n')
+md_output = ''
+
+i=0
+
+while (i<len(lines)):
+	
+	#append all following lines of a table row
+	if (lines[i][:1]=='|'):
+		while(i<len(lines) and lines[i][-1:]!='|'):
```
Comment (zonradkuse) [inline md_format_tables.py]: this iteration is crucial - maybe add a comment what it does - i.e. just elaborate a bit how this is working :)

**Review by zonradkuse — COMMENTED** [review]
:+1:

**Comment by wanLo** [comment]
> Does this occur at all (cells starting with line breaks)? Probably not that important though man_shrugging

Thought about counting the occurrences of | in the (non-broken) table header. But taking a closer look, this doesn't happen at all - cell borders and cell content ar always separated with white space. Tables broken to this extend wouldn't be this easy to fix automatically.

File: md_format_tables.py (hunk 1 above)
Comment (wanLo) [reply md_format_tables.py]: I refactored the code accordingly.

**Comment by marcelklehr** [comment]
So this script will be run on wiki pages that are still in markdown only? How many of those are there? I was under the impression the majority was already converted.

**Comment by zonradkuse** [comment]
Unfortunately, the contrary is true. Most pages are not converted. Only those which were edited by some member before and probably not even here all were converted.

**Comment by marcelklehr** [comment]
Ok, so part of the link-migration process is 

1) Fix markdown tables
2) Convert everything to xwiki syntax
3) Migrate linked documents

?

**Comment by zonradkuse** [comment]
We should swap step 2 and 3 in order to make sure nothing weird happens to the links during conversion. But besides that, this is the gameplan.
